homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -553,104 +520,109 @@
-        except DeviceException as ex:
-            raise PlatformNotReady from ex
-
-    if model in PURIFIER_MIOT:
-        air_purifier = AirPurifierMiot(host, token)
-        device = XiaomiAirPurifierMiot(name, air_purifier, model, unique_id)
-    elif model.startswith("zhimi.airpurifier."):
-        air_purifier = AirPurifier(host, token)
-        device = XiaomiAirPurifier(name, air_purifier, model, unique_id)
-    elif model in HUMIDIFIER_MIOT:
-        air_humidifier = AirHumidifierMiot(host, token)
-        device = XiaomiAirHumidifierMiot(name, air_humidifier, model, unique_id)
-    elif model.startswith("zhimi.humidifier."):
-        air_humidifier = AirHumidifier(host, token, model=model)
-        device = XiaomiAirHumidifier(name, air_humidifier, model, unique_id)
-    elif model.startswith("zhimi.airfresh."):
-        air_fresh = AirFresh(host, token)
-        device = XiaomiAirFresh(name, air_fresh, model, unique_id)
-    else:
-        _LOGGER.error(
-            "Unsupported device found! Please create an issue at "
-            "https://github.com/syssi/xiaomi_airpurifier/issues "
-            "and provide the following data: %s",
-            model,
+    """Import Miio configuration from YAML."""
+    _LOGGER.warning(
+        "Loading Xiaomi Miio Fan via platform setup is deprecated. Please remove it from your configuration."
+    )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
         )
-        return False
-
-    hass.data[DATA_KEY][host] = device
-    async_add_entities([device], update_before_add=True)
-
-    async def async_service_handler(service):
-        """Map services to methods on XiaomiAirPurifier."""
-        method = SERVICE_TO_METHOD.get(service.service)
-        params = {
-            key: value for key, value in service.data.items() if key != ATTR_ENTITY_ID
-        }
-        entity_ids = service.data.get(ATTR_ENTITY_ID)
-        if entity_ids:
-            devices = [
-                device
-                for device in hass.data[DATA_KEY].values()
-                if device.entity_id in entity_ids
-            ]
+    )
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Fan from a config entry."""
+    entities = []
+
+    if config_entry.data[CONF_FLOW_TYPE] == CONF_DEVICE:
+        if DATA_KEY not in hass.data:
+            hass.data[DATA_KEY] = {}
+
+        host = config_entry.data[CONF_HOST]
+        token = config_entry.data[CONF_TOKEN]
+        name = config_entry.title
+        model = config_entry.data[CONF_MODEL]
+        unique_id = config_entry.unique_id
+
+        _LOGGER.debug("Initializing with host %s (token %s...)", host, token[:5])
+
+        if model in MODELS_PURIFIER_MIOT:
+            air_purifier = AirPurifierMiot(host, token)
+            device = XiaomiAirPurifierMiot(name, air_purifier, config_entry, unique_id)
```

Comment:
Please name all the `Entity` instances `entity` instead.
```suggestion
            entity = XiaomiAirPurifierMiot(name, air_purifier, config_entry, unique_id)
```

Reply:
done